presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -661,6 +663,35 @@ private static SliceWriteFunction typedVarcharWriteFunction(String jdbcTypeName)
         };
     }
 
+    private static ColumnMapping moneyColumnMapping()
+    {
+        // Money mapping can be improved when PostgreSQL gains explicit money type support.
+        // See https://github.com/pgjdbc/pgjdbc/issues/425 for more information.
+        return ColumnMapping.sliceMapping(
+                VARCHAR,
+                new SliceReadFunction()
+                {
+                    @Override
+                    public boolean isNull(ResultSet resultSet, int columnIndex)
+                            throws SQLException
+                    {
+                        // In PostgreSQL JDBC, money is mapped as double.
+                        // .getObject() calls .getDouble and it fails to parse the money value.
+                        resultSet.getString(columnIndex);
+                        return resultSet.wasNull();
+                    }
+
+                    @Override
+                    public Slice readSlice(ResultSet resultSet, int columnIndex)
+                            throws SQLException
+                    {
+                        return utf8Slice(resultSet.getString(columnIndex));
+                    }
+                },
+                (statement, index, value) -> { throw new PrestoException(NOT_SUPPORTED, "Money type is not supported for INSERT"); },
```

Comment:
Is there any reason not doing the following?
```java
typedVarcharWriteFunction("money")
```

Reply:
No, i didn't, thanks for point this out.

I don't fully understand the rules of the money type in pg. 
I sympathize a bit with https://github.com/pgjdbc/pgjdbc/issues/425#issuecomment-534990746 .
I'd prefer to understand the type better before unlocking writing.

Also, current mapping `money → varchar` isn't really suitable for _writing_.
Mapping to `decimal(.., 2)` sounds like a better idea, but this is much harder to implement the reading.
Postgres returns a String like `$10,000.42` and I'd need to parse it. OK, but what do i do if it returns `¥10,000.42`?
That's why i mapped it to `varchar`, without any interpretation on our side.

Reply:
I added some comment in the code.